### PR TITLE
Fix issues after 0.14.0 sync

### PR DESCRIPTION
**Pull request by Jooho**
**What this PR does / why we need it**:
- remove patch for `webhookconfiguration` to fix kustomize build issue.
- fix e2e tests for `predictor`, `path_based_routing`, `raw`

**Which issue(s) this PR fixes** *(optional, in `fixes #<issue number>(, fixes #<issue_number>, ...)` format, will close the issue(s) when PR gets merged)*:
Fixes #

**Type of changes**
Please delete options that are not relevant.

- [X] Bug fix (non-breaking change which fixes an issue)
- [ ] New feature (non-breaking change which adds functionality)
- [ ] Breaking change (fix or feature that would cause existing functionality to not work as expected)
- [ ] This change requires a documentation update

**Feature/Issue validation/testing**:

Please describe the tests that you ran to verify your changes and relevant result summary. Provide instructions so it can be reproduced.
Please also list any relevant details for your test configuration.

- [X] kustomize issue
```
kustomize build ./config/overlay/odh
kustomize build ./config/overlay/test
```

- [X] e2e test
~~~
 ./test/scripts/openshift-ci/run-e2e-tests.sh path_based_routing
 ./test/scripts/openshift-ci/run-e2e-tests.sh predictor ./test/scripts/openshift-ci/run-e2e-tests.sh predictor
 ./test/scripts/openshift-ci/run-e2e-tests.sh raw
~~~
- Logs

**Special notes for your reviewer**:

1. Please confirm that if this PR changes any image versions, then that's the sole change this PR makes.

**Checklist**:

- [X] Have you added unit/e2e tests that prove your fix is effective or that this feature works?
- [X] Has code been commented, particularly in hard-to-understand areas?
- [ ] Have you made corresponding changes to the documentation?

**Release note**:

```release-note

```

**Re-running failed tests**

- `/rerun-all` - rerun all failed workflows.
- `/rerun-workflow <workflow name>` - rerun a specific failed workflow. Only one workflow name can be specified. Multiple /rerun-workflow commands are allowed per comment.

### CODE REVIEW

File: config/crd/external/datasciencecluster.opendatahub.io_datascienceclusters.yaml [hunk 1]
```diff
@@ -0,0 +1,763 @@
+---
+apiVersion: apiextensions.k8s.io/v1
```
Comment (spolti) [inline config/crd/external/datasciencecluster.opendatahub.io_datascienceclusters.yaml:2]: why is this needed?

File: test/scripts/gh-actions/run-e2e-tests.sh [hunk 2]
```diff
@@ -28,7 +28,7 @@ else
 fi
 
 PARALLELISM="${2:-1}"
-source python/kserve/.venv/bin/activate
+#source python/kserve/.venv/bin/activate
```
Comment (spolti) [inline test/scripts/gh-actions/run-e2e-tests.sh:31]: is this intended?
IIRC the dependencies are installed using this venv.

File: test/scripts/openshift-ci/run-e2e-tests.sh [hunk 3]
```diff
@@ -71,19 +71,24 @@ if [ "$1" != "raw" ]; then
...
-# Patch the inferenceservice-config ConfigMap, when running RawDeployment tests
- if [ "$1" == "raw" ]; then
+kustomize build $PROJECT_ROOT/config/overlays/test |
+  sed "s|kserve/storage-initializer:latest|${STORAGE_INITIALIZER_IMAGE}|" |
+  sed "s|kserve/agent:latest|${KSERVE_AGENT_IMAGE}|" |
+  sed "s|kserve/router:latest|${KSERVE_ROUTER_IMAGE}|" |
+  sed "s|kserve/kserve-controller:latest|${KSERVE_CONTROLLER_IMAGE}|" |
+  oc apply --server-side=true -f -
+
+Patch the inferenceservice-config ConfigMap, when running RawDeployment tests
```
Comment (spolti) [inline test/scripts/openshift-ci/run-e2e-tests.sh:81]: Isn't it a comment?

File: test/scripts/openshift-ci/run-e2e-tests.sh [hunk 4]
```diff
@@ -71,19 +71,24 @@ if [ "$1" != "raw" ]; then
...
+if [ "$1" == "raw" ]; then
   export OPENSHIFT_INGRESS_DOMAIN=$(oc get ingresses.config cluster -o jsonpath='{.spec.domain}')
   oc patch configmap inferenceservice-config -n kserve --patch-file <(cat config/overlays/test/configmap/inferenceservice-openshift-ci-raw.yaml | envsubst)
   oc delete pod -n kserve -l control-plane=kserve-controller-manager
- fi
+
+  oc patch DataScienceCluster test-dsc --type='json' -p='[{"op": "replace", "path": "/spec/components/kserve/defaultDeploymentMode", "value": "RawDeployment"}]'
+else
+  export OPENSHIFT_INGRESS_DOMAIN=$(oc get ingresses.config cluster -o jsonpath='{.spec.domain}')
+  oc patch configmap inferenceservice-config -n kserve --patch-file <(cat config/overlays/test/configmap/inferenceservice-openshift-ci-serverless.yaml | envsubst)
```
Comment (israel-hdez) [inline test/scripts/openshift-ci/run-e2e-tests.sh:90]: I'm confused about why you need to configure the ingress domain here.

In the [comment docs](https://github.com/kserve/kserve/blob/b4ce5d6ec506bd064fda87b230b25b23f21397c5/config/configmap/inferenceservice.yaml#L230-L233), it is stated that this is only relevant for `RawDeployment`, but Raw is covered in the other branch of the `if`

File: config/crd/external/datasciencecluster.opendatahub.io_datascienceclusters.yaml (hunk 1 above)
Comment (israel-hdez) [reply config/crd/external/datasciencecluster.opendatahub.io_datascienceclusters.yaml:2]: Yes. I also don't understand why this is needed.

**Review by israel-hdez — COMMENTED** [review]
This is deactivating several E2Es. Can we have the list of the ones that would be ran? (i.e. a screenshot of the PASSED ones)

File: config/crd/external/datasciencecluster.opendatahub.io_datascienceclusters.yaml (hunk 1 above)
Comment (Jooho) [reply config/crd/external/datasciencecluster.opendatahub.io_datascienceclusters.yaml:2]: because odh-model-controller look for this by default

File: test/scripts/gh-actions/run-e2e-tests.sh (hunk 2 above)
Comment (Jooho) [reply test/scripts/gh-actions/run-e2e-tests.sh:31]: for ci, I didn't check but from the script, it does not setup venv, so it failed on local

File: test/scripts/openshift-ci/run-e2e-tests.sh (hunk 3 above)
Comment (Jooho) [reply test/scripts/openshift-ci/run-e2e-tests.sh:81]: oops updated.

File: test/scripts/gh-actions/run-e2e-tests.sh (hunk 2 above)
Comment (Jooho) [reply test/scripts/gh-actions/run-e2e-tests.sh:31]: updated

**Review by israel-hdez — APPROVED** [review]
